contracts/global/PriceFeedStore.sol

```diff
@@ -44,49 +42,77 @@ contract PriceFeedStore is PriceFeedValidationTrait, AuditManager, IPriceFeedSto
     mapping(address => PriceFeedInfo) public priceFeedInfo;
 
     /// @notice Returns the list of price feeds available for a token
-    function getPriceFeeds(address token) external view returns (address[] memory priceFeeds) {
-        return _allowedPriceFeeds[token].values();
+    function getPriceFeeds(address token) external view returns (address[] memory) {
+        address[] memory priceFeeds = _allowedPriceFeeds[token].values();
+        address[] memory equivalentPriceFeeds = _allowedPriceFeeds[equivalentTokens[token]].values();
+
+        return _mergeArrays(priceFeeds, equivalentPriceFeeds);
+    }
+
+    /// @dev Merges two address arrays so that there are no repetitions
+    function _mergeArrays(address[] memory a0, address[] memory a1) internal pure returns (address[] memory result) {
+        uint256 len0 = a0.length;
+        uint256 len1 = a1.length;
+
+        address[] memory _res = new address[](len0 + len1);
+
+        for (uint256 i = 0; i < len0;) {
+            _res[i] = a0[i];
+
+            unchecked {
+                ++i;
+            }
+        }
+
+        uint256 k = len0;
+
+        for (uint256 i = 0; i < len1;) {
+            for (uint256 j = 0; j <= k;) {
+                if (j == k) {
+                    _res[k] = a1[i];
+                    ++k;
+                    break;
+                }
+
+                if (_res[j] == a1[i]) break;
+
+                unchecked {
+                    ++j;
+                }
+            }
+
+            unchecked {
+                ++i;
+            }
+        }
+
+        result = new address[](k);
+
+        for (uint256 i = 0; i < k;) {
+            result[i] = _res[i];
+
+            unchecked {
+                ++i;
+            }
+        }
     }
 
     /// @notice Returns whether a price feed is allowed to be used for a token
     function isAllowedPriceFeed(address token, address priceFeed) external view returns (bool) {
-        return _priceFeedVerified(priceFeed)
-            && (
-                _allowedPriceFeeds[equivalentTokens[token]].contains(priceFeed)
-                    || _allowedPriceFeeds[token].contains(priceFeed)
-            );
+        return _allowedPriceFeeds[equivalentTokens[token]].contains(priceFeed)
+            || _allowedPriceFeeds[token].contains(priceFeed);
```

Comment:
though gas is not the main concern, it does seem that checking token's own price feeds first would be much cheaper on average